annotation-processor/src/main/java/org/hibernate/validator/ap/checks/MethodParameterOverrideCheck.java

```diff
@@ -0,0 +1,182 @@
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Checks, that the given parameter has a correct set of constraints if it's overriding a method form other class.
+ * Applies to methods parameters.
+ */
+public class MethodParameterOverrideCheck extends AbstractConstraintCheck {
+
+	private static final String JAVA_LANG_OBJECT = "java.lang.Object";
+
+	private Elements elementUtils;
+
+	private Types typeUtils;
+
+	public MethodParameterOverrideCheck(Elements elementUtils, Types typeUtils) {
+		this.elementUtils = elementUtils;
+		this.typeUtils = typeUtils;
+	}
+
+	@Override
+	public Set<ConstraintCheckError> checkField(VariableElement element, AnnotationMirror annotation) {
+		// get a current class in which a method is located.
+		ExecutableElement currentMethod = (ExecutableElement) element.getEnclosingElement();
+		TypeElement currentTypeElement = (TypeElement) typeUtils.asElement( currentMethod.getEnclosingElement().asType() );
+
+		// find if there's a method that was overridden by a current one.
+		List<ExecutableElement> overridden = findAllOverriddenElements( currentTypeElement, currentMethod );
+		if ( overridden.isEmpty() ) {
+			return Collections.emptySet();
+		}
+
+		// if there's more than one overridden method we need to make sure that all of them match
+		Optional<ConstraintCheckError> error = overridden.stream()
+				.flatMap(
+						overriddenElement -> overridden.stream()
+								.filter( otherElement -> !otherElement.equals( overriddenElement ) )
+								.map( otherElement -> !correctlyOverrides( overriddenElement, otherElement ) ? new ConstraintCheckError( element, annotation, "INCORRECT_METHOD_OVERRIDING" ) : null )
+								.filter( constraintCheckError -> constraintCheckError != null )
+				).findAny();
+
+		if ( error.isPresent() ) {
+			return CollectionHelper.asSet( error.get() );
+		}
+
+		// if we reached this part of code it means we need to check if the current method 'correctly' overrides super methods.
+		for ( ExecutableElement overriddenExecutableElement : overridden ) {
+			if ( !correctlyOverrides( currentMethod, overriddenExecutableElement ) ) {
+				return CollectionHelper.asSet( new ConstraintCheckError( element, annotation, "INCORRECT_METHOD_OVERRIDING" ) );
+
+			}
+		}
+
+		return Collections.emptySet();
+	}
+
+	/**
+	 * Determine if one method 'correctly' overrides another one
+	 *
+	 * @param current    method from a current subclass
+	 * @param overridden method from a super class
+	 * @return {@code true} if method is overridden 'correctly', {@code false} otherwise
+	 */
+	private boolean correctlyOverrides(ExecutableElement current, ExecutableElement overridden) {
```

Comment:
yes, I've used this method in `getOverriddenElement` at line 178 to filter out methods that are not overriding a method under validation. And as for this private method `correctlyOverrides` it's verifying that the annotations are correctly used on current method and the one that was overridden, as `Elements#overrides()` seems to ignore annotations. So this `correctlyOverrides` consumes two methods - overridden and overriding.